src/cript/nodes/core.py

```diff
@@ -91,20 +147,33 @@ def _from_json(cls, json_dict: dict):
         # We create manually a dict that contains all elements from the send dict.
         # That eliminates additional fields and doesn't require asdict.
         arguments = {}
-        for field in cls.JsonAttributes().__dataclass_fields__:
-            if field in json_dict:
+        default_dataclass = cls.JsonAttributes()
+        for field in json_dict:
+            try:
+                getattr(default_dataclass, field)
+            except AttributeError:
+                pass
+            else:
                 arguments[field] = json_dict[field]
 
         # The call to the constructor might ignore fields that are usually not writable.
         try:
             node = cls(**arguments)
+        # TODO we should not catch all exceptions if we are handling them, and instead let it fail
+        #  to create a good error message that points to the correct place that it failed to make debugging easier
         except Exception as exc:
             print(cls, arguments)
             raise exc
-        attrs = cls.JsonAttributes(**arguments)
-        # Handle UID manually. Conserve newly assigned uid if uid is default (empty)
-        if attrs.uid == cls.JsonAttributes().uid:
-            attrs = replace(attrs, uid=node.uid)
+
+        try:
+            attrs = cls.JsonAttributes(**arguments)
+        except TypeError:
+            help(cls.JsonAttributes)
```

Comment:
do we need to have this here or was this left by mistake?

Reply:
good catch, this has to go.